src/nvim/buffer_defs.h

```diff
@@ -1101,6 +1099,24 @@ typedef struct {
   pos_T w_cursor_corr;  // corrected cursor position
 } pos_save_T;
 
+/// Status line click definition
+typedef struct {
+  enum {
+    kStlClickDisabled = 0,  ///< Clicks to this area are ignored.
+    kStlClickTabSwitch,     ///< Switch to the given tab.
+    kStlClickTabClose,      ///< Close given tab.
+    kStlClickFuncRun,       ///< Run user function.
+  } type;      ///< Type of the click.
+  int tabnr;   ///< Tab page number.
+  char *func;  ///< Function to run.
+} StlClickDefinition;
+
+/// Used for tabline clicks
+typedef struct {
+  StlClickDefinition def;  ///< Click definition.
+  const char *start;       ///< Location where region starts.
+} StlClickRecord;
```

Comment:
I first tried moving these and the `tab_page_click_defs(_size)` globals to `statusline.h`, but ran into define/linking issues. Moving them to `buffer_defs.h`( where they are used in the window structure), and `globals.h`(where other tabline globals live) seems fine as well.

Reply:
do we need `statusline_defs.h` ?